warehouse/http.go

```diff
@@ -0,0 +1,404 @@
+	}
+	a.config.healthTimeout = conf.GetDuration("Warehouse.healthTimeout", 10, time.Second)
+	a.config.readerHeaderTimeout = conf.GetDuration("Warehouse.readerHeaderTimeout", 3, time.Second)
+	a.config.runningMode = conf.GetString("Warehouse.runningMode", "")
+	a.config.webPort = conf.GetInt("Warehouse.webPort", 8082)
+
+	return a
+}
+
+func (a *Api) Start(ctx context.Context) error {
+	if !isStandAlone(a.mode) {
+		return nil
+	}
+
+	srvMux := chi.NewRouter()
+	srvMux.Get("/health", a.healthHandler)
+
+	if a.config.runningMode != DegradedMode {
+		if isMaster(a.mode) {
+			a.addWarehouseEndpoints(ctx, srvMux)
+
+			a.logger.Infof("Starting warehouse master service in %d", a.config.webPort)
+		} else {
+			a.logger.Infof("Starting warehouse slave service in %d", a.config.webPort)
+		}
+	}
+
+	srv := &http.Server{
+		Addr:              fmt.Sprintf(":%d", a.config.webPort),
+		Handler:           bugsnag.Handler(srvMux),
+		ReadHeaderTimeout: a.config.readerHeaderTimeout,
+	}
+	return kithttputil.ListenAndServe(ctx, srv)
+}
+
+func (a *Api) addWarehouseEndpoints(ctx context.Context, mux *chi.Mux) {
+	a.logger.Infof("waiting for BackendConfig before starting on %d", a.config.webPort)
+
+	a.bcConfig.WaitForConfig(ctx)
+
+	// process the staging file
+	mux.Handle("/v1/process", (&api.WarehouseAPI{
+		Logger:      a.logger,
+		Stats:       a.statsFactory,
+		Repo:        a.stagingRepo,
+		Multitenant: a.tenantManager,
+	}).Handler())
+
+	// triggers upload only when there are pending events and triggerUpload is sent for a sourceId
+	mux.Post("/v1/warehouse/pending-events", a.pendingEventsHandler)
+
+	// triggers uploads for a source
+	mux.Post("/v1/warehouse/trigger-upload", a.triggerUploadHandler)
+
+	// Warehouse Async Job end-points
+	mux.Post("/v1/warehouse/jobs", a.asyncManager.InsertJobHandler)       // FIXME: add degraded mode
+	mux.Get("/v1/warehouse/jobs/status", a.asyncManager.StatusJobHandler) // FIXME: add degraded mode
+
+	// TODO: Remove this endpoint once sources change is released
+	// fetch schema info
+	mux.Get("/v1/warehouse/fetch-tables", a.fetchTablesHandler)
+	mux.Get("/internal/v1/warehouse/fetch-tables", a.fetchTablesHandler)
+}
+
+func (a *Api) healthHandler(w http.ResponseWriter, r *http.Request) {
+	var dbService, pgNotifierService string
+
+	ctx, cancel := context.WithTimeout(r.Context(), a.config.healthTimeout)
+	defer cancel()
+
+	if a.config.runningMode != DegradedMode {
+		if !checkHealth(ctx, a.notifier.GetDBHandle()) {
+			http.Error(w, "Cannot connect to pgNotifierService", http.StatusInternalServerError)
+			return
+		}
+		pgNotifierService = "UP"
+	}
+
+	if isMaster(a.mode) {
+		if !checkHealth(ctx, a.db.DB) {
+			http.Error(w, "Cannot connect to dbService", http.StatusInternalServerError)
+			return
+		}
+		dbService = "UP"
+	}
+
+	healthVal := fmt.Sprintf(`
+{
+	"server": "UP",
+	"db": %q,
+	"pgNotifier": %q,
+	"acceptingEvents": "TRUE",
+	"warehouseMode": %q,
+	"goroutines": "%d"
```

Comment:
@fracasula @lvrach 
Should we remove `goroutines` from the health payload? Since we removed it from the rudder server because of [security concerns](https://github.com/rudderlabs/rudder-server/issues/2075).

Reply:
Also, I don't see any point in exposing it.

Reply:
Yes we can remove them. If we don't have the debug/pprof endpoint then we can create a task to add it.

Reply:
We have pprof endpoint.